source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -0,0 +1,187 @@
+#include "AMGenericStepScanConfiguration.h"
+
+#include "ui/acquaman/AMGenericStepScanConfigurationView.h"
+#include "acquaman/AMGenericStepScanController.h"
+
+#include <math.h>
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(QObject *parent)
+	: AMStepScanConfiguration(parent)
+{
+	setName("Generic Scan");
+	setUserScanName("Generic Scan");
+}
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(const AMGenericStepScanConfiguration &original)
+	: AMStepScanConfiguration(original)
+{
+	setName(original.name());
+	setUserScanName(original.name());
+
+	computeTotalTime();
+
+	for (int i = 0, size = scanAxes_.count(); i < size; i++){
```

Comment:
Yes, this is done on purpose.  This is one of those trade offs where coding efficiency is traded for run time efficiency.  In general, I will _always_ go for run time efficiency because that is one of the limitations that we can actually run up against.